src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -124,28 +125,22 @@ async Task IKernelCommandHandler<SendValue>.HandleAsync(SendValue command, Kerne
 
     private Task SetValueAsync(string valueName, object value, Type? declaredType = null)
     {
-        if(value is (HttpVariableDeclarationAndAssignmentNode))
-        {
-            var variable = (HttpVariableDeclarationAndAssignmentNode) value;
-            if(variable.ValueNode is not null)
-            {
-                var bindingResult = variable.ValueNode.TryGetValue(BindExpressionValues);
-                if (bindingResult.IsSuccessful && bindingResult.Value is not null)
-                {
-                    _variables[valueName] = bindingResult.Value;
-                }
-            }
-            
-        } 
-        else
-        {
-            _variables[valueName] = value;
-        }
+        _variables[valueName] = value;
         return Task.CompletedTask;
     }
 
     async Task IKernelCommandHandler<SubmitCode>.HandleAsync(SubmitCode command, KernelInvocationContext context)
     {
+        string? doc = "";
+        if (command.Parameters.TryGetValue("Document", out doc))
```

Comment:
How are we planning to make this would work in the case of notebooks in VS Code or the dotnet-repl tool? Would we need changes for those clients to repeat the notebook cell contents in both 'Document' parameter as well as in `SubmitCode.Code` to ensure that any variables included in the code are parsed and bound?

Do we need code in the http kernel above to also parse and bind variables present in the `SubmitCode.Code` (in addition to the `Document` parameter) to support the notebooks and dotnet-repl scenarios?

Reply:
Adding a property to the kernel that tracks the current document and if there are any changes from it then we'll go ahead with the re-parse

Reply:
This has the potential to introduce bugs where the kernel state gets out of sync with the editor state. What's the performance impact of resending and reparsing the whole document?